quadratic/quadratic.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+/*Given coefficients A, B, and C of a quadratic equation Ax^2 + Bx + C = 0, write a program that prints the equation solutions x1 and x2.
+Now program also can solve linear equation. Oh.
+You will need to use math.Sqrt function to calculate a square root.
+Example:
+The program finds solutions of a quadratic equation.
+Enter the coefficients A, B and C:
+1 3 -10
+x1 = 2
+x2 = -5 */
+func main() {
+	fmt.Println("The program finds solutions of a quadratic equation.")
+	fmt.Println("Enter the coefficients A, B and C:")
+	var A, B, C float64
+	fmt.Scan(&A, &B, &C)
+	var D float64 = B*B - 4*A*C
+	if A == 0 {
+		fmt.Println("Its linear equation!")
+		if B != 0 {
+			var lx float64 = -C / B
+			fmt.Printf("x: %v", lx)
+		}
+		if B == 0 {
+			fmt.Println("No answer.")
+		}
+	}
+	if A != 0 {
```

Comment:
Please use `else` statements (in other places as well):
```
if condition {
  // ...
} else {
  // ...
}
```

Reply:
Since the `A == 0` block is relatively large, I recommend to use early return instead of having nested code:
```
if A == 0 {
  // handle linear equation Bx+C=0
  // ...
  return // <-- Here's that early return.
}
// handle square equation Ax^2+Bx+C=0
// no need to handle A != 0, since when we get here A cannot be 0.
```